### PR TITLE
framework: bump kernel version

### PR DESCRIPTION
**Pull request by Mic92**
The current lts (5.15) also did not work with this wifi chip. I have not tested 5.16 but 5.17 works for me.

### CODE REVIEW

**Comment by Mic92** [comment]
cc @Emiller88

**Comment by 0x4A6F** [comment]
I'm on 5.16 and it has working wifi.

**Comment by Emiller88** [comment]
`5.15.11` still works for me. But that's fine with me if you want to bump it, I just don't have a quick way to test it.

File: framework/default.nix [hunk 1]
```diff
@@ -9,9 +9,9 @@
   # https://kvark.github.io/linux/framework/2021/10/17/framework-nixos.html
   boot.kernelParams = [ "mem_sleep_default=deep" ];
 
-  # Requires at least 5.12 for working wi-fi and bluetooth.
+  # Requires at least 5.17 for working wi-fi and bluetooth.
   # https://community.frame.work/t/using-the-ax210-with-linux-on-the-framework-laptop/1844/89
-  boot.kernelPackages = lib.mkIf (lib.versionOlder pkgs.linux.version "5.12") (lib.mkDefault pkgs.linuxPackages_latest);
+  boot.kernelPackages = lib.mkIf (lib.versionOlder pkgs.linux.version "5.17") (lib.mkDefault pkgs.linuxPackages_latest);
```
Comment (Mic92) [inline framework/default.nix:14]:
```suggestion
  boot.kernelPackages = lib.mkIf (lib.versionOlder pkgs.linux.version "5.16") (lib.mkDefault pkgs.linuxPackages_latest);
```

**Comment by Mic92** [comment]
I also upgraded all the firmware... Maybe this could cause also troubles, don't know?

**Comment by Emiller88** [comment]
Ah, I haven't done that, so that could be it. Are there docs on how to upgrade the firmware? I'll sacrifice mine for the sake of community.

**Comment by Mic92** [comment]
First put enable `fwupd`

```
services.fwupd.enable = true;
```

Then run 

```
 $ sudo fwupdmgr update
```

I already went through this process.

**Pull request merged**